src/ostorlab/runtimes/local/runtime.py

```diff
@@ -297,9 +312,88 @@ def stop(self, scan_id: str) -> None:
             else:
                 console.info(f"Scan {scan_id} was not found.")
 
-    def _create_scan_db(self, title: str, asset: str):
+    def _create_scan_db(
+        self,
+        title: str,
+        asset: str,
+        assets: Optional[List[base_asset.Asset]],
+        agent_group_definition: definitions.AgentGroupDefinition,
+    ) -> models.Scan:
         """Persist the scan in the database"""
-        return models.Scan.create(title=title, asset=asset)
+        agent_group = self._create_agent_group_db(agent_group_definition)
+
+        scan = models.Scan.create(
+            title=title, asset=asset, agent_group_id=agent_group.id
+        )
+
+        self._create_assets_db(scan, assets)
+
+        return scan
+
+    def _create_assets_db(
+        self, scan: models.Scan, assets: Optional[List[base_asset.Asset]]
+    ) -> None:
+        """Persist the assets in the database."""
+        networks: List[str] = []
+        links = []
+        for asset in assets:
+            if (
+                isinstance(asset, ip.IP)
+                or isinstance(asset, ipv4.IPv4)
+                or isinstance(asset, ipv6.IPv6)
+            ):
+                networks.append(f"{asset.host}/{asset.mask}")
+            elif isinstance(asset, link.Link):
+                links.append(f'{{"url": "{asset.url}", "method": "{asset.method}"}}')
+            elif isinstance(asset, domain_name.DomainName):
+                links.append(asset.name)
+            elif isinstance(asset, ios_ipa.IOSIpa):
+                models.IosFile.create(path=asset.path, scan_id=scan.id)
+            elif isinstance(asset, ios_store.IOSStore):
+                models.IosStore.create(bundle_id=asset.bundle_id, scan_id=scan.id)
+            elif isinstance(asset, android_aab.AndroidAab) or isinstance(
+                asset, android_apk.AndroidApk
+            ):
+                models.AndroidFile.create(path=asset.path, scan_id=scan.id)
+            elif isinstance(asset, android_store.AndroidStore):
+                models.AndroidStore.create(
+                    package_name=asset.package_name, scan_id=scan.id
+                )
```

Comment:
else not implemented with a message indicating the asset !!

Reply:
the validation is handled in the cli, it cant be an asset other that the ones we defined